src/index.js

```diff
@@ -29,17 +29,24 @@ class MediaQuery extends React.Component {
     values: {}
   }
 
-  state = { matches: false }
+  constructor(props) {
+    super(props)
+    this.buildMatcher(props)
+    this.state = {
+      matches: this._mql.matches
+    }
+  }
 
-  componentWillMount() {
-    this.updateQuery(this.props)
+  componentDidMount() {
+    this._mql.addListener(this.updateMatches)
+    this.updateMatches()
   }
 
-  componentWillReceiveProps(nextProps) {
+  UNSAFE_componentWillReceiveProps(nextProps) {
```

Comment:
If we store `this._mql` in state, shouldn't we be able to use getDerivedStateFromProps to manage updating it whenever the props change?

Reply:
I think that could work. Should we put that change into this pr?

Reply:
@whatknight Yeah, I think we can make this a full "modernize to latest react conventions" PR

Reply:
Sounds good!